fastcrypto/src/encoding.rs

```diff
@@ -266,3 +267,22 @@ impl<'de, const N: usize> DeserializeAs<'de, [u8; N]> for Base58 {
         Ok(array)
     }
 }
+
+/// Bech32 encoding
+pub struct Bech32(String);
+
+impl Bech32 {
+    pub fn decode(s: &str, hrp: &str) -> Result<Vec<u8>, eyre::Report> {
+        let (parsed, data, variant) = bech32::decode(s).map_err(|e| eyre::eyre!(e))?;
+        if parsed != hrp || variant != Variant::Bech32 {
+            Err(eyre!("invalid hrp or variant"))
+        } else {
+            Vec::<u8>::from_base32(&data).map_err(|e| eyre::eyre!(e))
+        }
+    }
+
+    pub fn encode<T: AsRef<[u8]>>(data: T, hrp: &str) -> Result<String> {
```

Comment:
Since it uses AsRef, data can be a ref.
(in general we should reduce dependency on AsRef for fastcrypt types, and move to non ref byte arrays / serialized objects, but we can do that later)